test/DefaultCluster.Tests/LifecycleObserverCreationTests.cs

```diff
@@ -0,0 +1,159 @@
+            {
+                _grainFactory = grainFactory;
+            }
+
+            public void Participate(ISiloLifecycle lifecycle)
+            {
+                lifecycle.Subscribe(
+                    nameof(TestLifecycleParticipant),
+                    ServiceLifecycleStage.Active,
+                    OnStart,
+                    OnStop);
+            }
+
+            private Task OnStart(CancellationToken ct)
+            {
+                // This is the critical test - creating a grain observer during lifecycle participation
+                // should work now that lifecycle events run via Task.Run instead of on a SystemTarget
+                var observer = new TestObserver();
+                var reference = _grainFactory.CreateObjectReference<ISimpleGrainObserver>(observer);
+                
+                // Verify the reference was created successfully
+                Assert.NotNull(reference);
+                
+                // Clean up
+                _grainFactory.DeleteObjectReference<ISimpleGrainObserver>(reference);
+                
+                Fixture.ObserverCreated = true;
+                return Task.CompletedTask;
+            }
+
+            private Task OnStop(CancellationToken ct)
+            {
+                return Task.CompletedTask;
+            }
+        }
+
+        /// <summary>
+        /// Simple observer implementation for testing.
+        /// </summary>
+        private class TestObserver : ISimpleGrainObserver
+        {
+            public void StateChanged(int a, int b)
+            {
+                // No-op for testing
+            }
+        }
+
+        public LifecycleObserverCreationTests(Fixture fixture)
+        {
+            _host = fixture.Host;
+        }
+
+        /// <summary>
+        /// Tests that grain observers can be created during lifecycle participation.
+        /// This validates the fix for the issue where CreateObjectReference would fail
+        /// with "Cannot create a local object reference from a grain" when called during
+        /// lifecycle events, because those events were previously scheduled on a SystemTarget.
+        /// </summary>
+        [Fact]
+        public void LifecycleParticipant_CanCreateGrainObserver()
+        {
+            // The test passes if the host started successfully and the observer was created
+            // in the TestLifecycleParticipant.OnStart method without throwing an exception
+            Assert.True(Fixture.ObserverCreated, "Observer should have been created during lifecycle participation");
+        }
+
+        /// <summary>
+        /// Tests that grains can be called from within lifecycle participants.
+        /// This further validates that lifecycle events now run without a grain context,
+        /// enabling normal grain calls and observer creation.
+        /// </summary>
+        [Fact]
+        public async Task LifecycleParticipant_CanCallGrains()
```

Comment:
This is not calling a grain from within a lifecycle participant - it's calling the grain once the cluster has already fully started.